lib/websocket_adapter/standard.rb

```diff
@@ -0,0 +1,40 @@
+module WebsocketAdapter
+  class Standard
+    attr_reader :env, :url
+
+    def initialize(env, sock)
+      @env = env
+      @sock = sock
+
+      secure = Rack::Request.new(env).ssl?
+      scheme = secure ? 'wss:' : 'ws:'
+      @url = scheme + '//' + env['HTTP_HOST'] + env['REQUEST_URI']
```

Comment:
Be VERY careful about URI building with hostnames... you probably want to use URI::Generic(scheme => schem).build(...)  (see also [this](https://github.com/ruby/ruby/pull/765)), examples are [here](https://github.com/ManageIQ/manageiq/pull/7115) and other places too but that should be enough.

How does this handle ipv6?  Is HTTP_HOST `::1` or `[::1]`?  I think it's the first if it's used for socket connections.  If you use URI::Generic.build, it just works and you don't have to worry about wrapping/unwrapping the `[]` for ipv6 adddresses.

Reply:
Sure, add a todo and come back to it... I can't promise I'll see it next time so I wanted to raise it now... I spent many months on ipv6 in our code and libraries so I'm permanently scarred.

Reply:
Added a FIXME :wink: